### PR TITLE
boot_vm_in_hugepage.1G_hugepage: add new case for 1G hugepage, boot, reboot, shutdown

### PR DESCRIPTION
**Pull request by ybduan**
boot_vm_in_hugepage.1G_hugepage:
    Update previous scenario configurations and move it in variant "2M_hugepage"
    Add new case for 1G hugepage, boot, reboot, shutdown

ID: 1467177

Signed-off-by: yduan <yduan@redhat.com>

### CODE REVIEW

**Comment by ybduan** [comment]
(1/1) smp_16.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_vm_in_hugepage.1G_hugepage: PASS (431.74 s)

File: generic/tests/cfg/boot_vm_in_hugepage.cfg [hunk 1]
```diff
@@ -5,12 +5,21 @@
...
+            backend_mem = memory-backend-file
+            mem-path = /mnt/kvm_hugepage
+            pre_command_noncritical = yes
+            pre_command = "echo 3 > /proc/sys/vm/drop_caches"
+        - 1G_hugepage:
+            # Notes:
+            #    Before start testing, please ensure your host OS support 1G hugepage.
+            #    Please don't forget to update host kernel line to enable 1G hugepage
+            #    support and ensure your host have enough memory to create guest memory.
+            extra_params += " -mem-path /mnt/kvm_hugepage"
```
Comment (yanan-fu) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: mem-path is common,  move out of the variants is ok.

File: generic/tests/cfg/boot_vm_in_hugepage.cfg [hunk 2]
```diff
@@ -5,12 +5,21 @@
...
+            excepted_hugepage_size = 1G
```
Comment (yanan-fu) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: expected_hugepage_size

File: generic/tests/cfg/boot_vm_in_hugepage.cfg [hunk 3]
```diff
@@ -5,12 +5,21 @@
...
-    mem-path = /mnt/kvm_hugepage
-    pre_command_noncritical = yes
-    pre_command = "echo 3 > /proc/sys/vm/drop_caches"
+    variants:
+        - default_2M_hugepage:
+            slots_mem = 4
+            maxmem_mem = 32G
+            mem_devs = hugepage
+            size_mem_hugepage = 256M
+            backend_mem = memory-backend-file
```
Comment (yanan-fu) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: Refer the comment from huangyum in https://github.com/autotest/tp-qemu/pull/1038:
"For this test, I don't think a pc-dimm is needed. Current qemu cmdline is OK."
No need to add backend_mem here ?

File: generic/tests/cfg/boot_vm_in_hugepage.cfg (hunk 2 above)
Comment (ybduan) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: Modified, Thanks!

**Comment by ybduan** [comment]
(1/1) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_vm_in_hugepage.2M_hugepage: PASS (198.77 s)

File: generic/tests/cfg/boot_vm_in_hugepage.cfg (hunk 3 above)
Comment (ybduan) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: Modified, Thanks!

File: generic/tests/cfg/boot_vm_in_hugepage.cfg (hunk 1 above)
Comment (ybduan) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: Modified, Thanks!

File: generic/tests/cfg/boot_vm_in_hugepage.cfg [hunk 4]
```diff
@@ -5,12 +5,16 @@
     kill_vm_on_error = yes
     login_timeout = 240
     reboot_count = 25
```
Comment (yanan-fu) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: Could we reduce the value of reboot_count ?   It is time-consuming.  IMHO, 10 times is enough.
And, it is not related to the new case in fact, just a suggestion, it is up to you.

**Comment by jingzhao84** [comment]
LGTM
smp_4.1024m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_vm_in_hugepage.1G_hugepage: PASS

File: generic/tests/cfg/boot_vm_in_hugepage.cfg [hunk 5]
```diff
@@ -5,12 +5,16 @@
...
-    slots_mem = 4
-    maxmem_mem = 32G
     setup_hugepages = yes
-    mem_devs = hugepage
-    size_mem_hugepage = 256M
-    backend_mem = memory-backend-file
```
Comment (CongLi) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: For the previous scenario, delete it directly?
no tests of the previous scenario?

it's better to add previous scenario as a variant.

**Comment by huangyum** [comment]
Since you modify the two cases actually, I think you need to run another case as well. 
It is better to run the case without specifying memory for guest.  
If the two cases can work well, I think the previous scenario is not necessary needed.

File: generic/tests/cfg/boot_vm_in_hugepage.cfg (hunk 5 above)
Comment (yanan-fu) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: @CongLi  I think it is ok to delete the previous scenario.  From @huangyum,  pc-dimm is not must for hugepage test,  and fixed slot and maxmem is inappropriate, current scenario is match with polarion case.

**Comment by CongLi** [comment]
If so, please make it clear in the commit log.

Thanks.

File: generic/tests/cfg/boot_vm_in_hugepage.cfg (hunk 4 above)
Comment (ybduan) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: I followed the former reboot_count. What's your opinion? @huangyum

Comment (huangyum) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: I suggest to follow our case step.

File: generic/tests/cfg/boot_vm_in_hugepage.cfg (hunk 5 above)
Comment (ybduan) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg]: I put the previous scenario in boot_vm_in_hugepage.2M_hugepage and changed the parameters according to feature owner's advice. Thanks!

**Comment by ybduan** [comment]
The test result of 2M hugepage had been put above. Thanks!

**Comment by ybduan** [comment]
Commit log has been updated, Could you please help to recheck? Thanks! @CongLi

**Comment by yanan-fu** [comment]
LGTM
Acked-by: Yanan Fu <yfu@redhat.com>

**Comment by CongLi** [comment]
@ybduan 
IMO, from your commit log, there should be another variant to cover boot, shutdown, reboot 3 scenarios,
but from your code, it only cover 'reboot' scenario.

Actually, it's not clear for me.

**Comment by ybduan** [comment]
Update the commit log again, Thank you! @CongLi

**Comment by huangyum** [comment]
Acked-by: Yumei Huang <yuhuang@redhat.com>

**Pull request merged**